### PR TITLE
Improve developer documentation

### LINKED ISSUE #305: Developer docs don't include instructions for running the testing framework locally, and also appear to be missing some setup steps.
I went to run pytest, and start experimenting with mutmut locally, and got this error:

```
(.venv) [esauer 🎩︎pelorus] (master)$ pytest
===================================================================================================== test session starts =====================================================================================================
platform linux -- Python 3.9.4, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/esauer/src/pelorus
plugins: pylava-0.3.0
collected 25 items / 2 errors / 23 selected                                                                                                                                                                                   

=========================================================================================================== ERRORS ============================================================================================================
____________________________________________________________ ERROR collecting samples/basic-nginx/galaxy/openshift-applier/molecule/default/tests/test_applier.py _____________________________________________________________
ImportError while importing test module '/home/esauer/src/pelorus/samples/basic-nginx/galaxy/openshift-applier/molecule/default/tests/test_applier.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
samples/basic-nginx/galaxy/openshift-applier/molecule/default/tests/test_applier.py:3: in <module>
    import testinfra.utils.ansible_runner
E   ModuleNotFoundError: No module named 'testinfra'
____________________________________________________ ERROR collecting samples/nginx-jenkins-binary-fromdir/galaxy/openshift-applier/molecule/default/tests/test_applier.py ____________________________________________________
ImportError while importing test module '/home/esauer/src/pelorus/samples/nginx-jenkins-binary-fromdir/galaxy/openshift-applier/molecule/default/tests/test_applier.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
samples/nginx-jenkins-binary-fromdir/galaxy/openshift-applier/molecule/default/tests/test_applier.py:3: in <module>
    import testinfra.utils.ansible_runner
E   ModuleNotFoundError: No module named 'testinfra'
=================================================================================================== short test summary info ===================================================================================================
ERROR samples/basic-nginx/galaxy/openshift-applier/molecule/default/tests/test_applier.py
ERROR samples/nginx-jenkins-binary-fromdir/galaxy/openshift-applier/molecule/default/tests/test_applier.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================================================================================== 2 errors in 8.86s ======================================================================================================

```

We should update the developer docs to provide the necessary steps to make tests locally runnable as part of development cycle.

### PR DESCRIPTION
**Pull request by mpryc**
Changes to the Development documentation to reflect current implementation.

Small change to the bump-version to make it work (tested on python 3.10.4).

Also Fixes #305

@redhat-cop/mdt

### CODE REVIEW

File: scripts/bump-version [hunk 1]
```diff
@@ -6,7 +6,7 @@ from pathlib import Path
 from lib.charts import bump
 
 parser = argparse.ArgumentParser(description="Bump versions in given helm charts")
-parser.add_argument("charts", metavar="chart", type=Path, nargs='+', help="The path to the Chart.yaml file, or the directory containing it", required=True)
+parser.add_argument("charts", metavar="chart", type=Path, nargs='+', help="The path to the Chart.yaml file, or the directory containing it")
```
Comment (weshayutin) [inline scripts/bump-version:9]: hrm.. maybe we can fix this before merging.
(.venv) [whayutin@thinkdoe pelorus]$ ./scripts/bump-version -help
Traceback (most recent call last):
  File "/home/whayutin/OPENSHIFT/git/KONVEYOR/PELORUS/upstream/pelorus/./scripts/bump-version", line 9, in <module>
    parser.add_argument("charts", metavar="chart", type=Path, nargs='+', help="The path to the Chart.yaml file, or the directory containing it", required=True)
  File "/usr/lib64/python3.10/argparse.py", line 1405, in add_argument
    kwargs = self._get_positional_kwargs(*args, **kwargs)
  File "/usr/lib64/python3.10/argparse.py", line 1521, in _get_positional_kwargs
    raise TypeError(msg)

**Review by weshayutin — CHANGES_REQUESTED** [review]
++ Everything looks awesome. Thanks for doc work! One minor thing

File: scripts/bump-version (hunk 1 above)
Comment (mpryc) [reply scripts/bump-version:9]: actually I fixed this error by this line :)

**Review by weshayutin — APPROVED** [review]

**Pull request merged**